torch/csrc/distributed/c10d/init.cpp

```diff
@@ -411,6 +411,42 @@ Example::
     >>> store.set("first_key", "first_value")
     >>> # Should return "first_value"
     >>> store.get("first_key")
+)")
+          .def(
+              "compare_set",
+              [](::c10d::Store& store,
+                 const std::string& key,
+                 const std::string& new_value,
+                 const std::string& old_value) -> py::bytes {
+                std::vector<uint8_t> newValue_(new_value.begin(), new_value.end());
+                std::vector<uint8_t> oldValue_(old_value.begin(), old_value.end());
+                auto value = store.compareSet(key, newValue_, oldValue_);
+                return py::bytes(
+                    reinterpret_cast<char*>(value.data()), value.size());
+              },
+              py::call_guard<py::gil_scoped_release>(),
+              R"(
+Inserts the key-value pair into the store based on the supplied ``key`` and
+performs comparison between ``new_value`` and ``old_value`` before inserting. ``new_value`` 
+will only be placed if ``old_value`` for the ``key`` already exists in the store.
+
+.. warning::
+    The ``compare_set`` API is only supported by the :class:`~torch.distributed.TCPStore`. Using this API
+    with the :class:`~torch.distributed.FileStore` or class:`~torch.distributed.HashStore` will result in an exception.
```

Comment:
In `PrefixStore` we should forward calls to `compare_set()` to the underlying store.

Reply:
Awesome catch, I totally forgot about prefixstore. Thank you Can!